Issue/quick 224 fix ci memory for real

dagguh opened the pull request.

mminns commented on virtual-users/src/main/kotlin/com/atlassian/performance/tools/hardware/vu/CustomScenario.kt:
```diff
@@ -34,7 +35,7 @@ class CustomScenario : Scenario {
     ): List<Action> {
         val drilldownMeter = meter.withW3cPerformanceTimeline(
-            JavascriptW3cPerformanceTimeline(jira.driver as JavascriptExecutor)
+            DisabledW3cPerformanceTimeline()
```
Doesn't this effect the _real_ test as well?


As I understand it provides client side timing statistics? Are they not needed?

dagguh replied:
You are very right. This does affect the real tests.

OTOH We weren't using waterfalls, so it doesn't deteriorate the use case.
If we need them back, we can have one "real" scenario and one "lightweight" scenario. Or we could find bigger agents or decrease the memory requirements in other ways.

dagguh committed:
QUICK-224: Point to Bamboo CI

dagguh committed:
QUICK-224: Reduce the post-processing memory footprint

Previously, the lock was attached to `HardwareExploration` instances,
which became concurrently executed with the parallelization of L and XL profiles.
Now the lock is JVM-wide, so it will be respected across threads.

dagguh committed:
QUICK-224: Fix a logging bug caused by Git sniffing

Fix errors like `ERROR StatusLogger No log4j2 configuration file found`.
It was caused by eager Git `Repository` class loading, which initializes a logger.
Then we obtain a workspace to put our custom loggers in, but log4j ignored our dynamic reconfig.

mminns committed:
QUICK-224: Add integration tests to verify HWR recommendation behaviour

dagguh committed:
QUICK-224: Fit in the CircleCI memory

The default `medium` CircleCI agent has 4 GB of RAM.
Reduce memory footprint by disabling drilldowns.
It affects not only the CI IT, but the actual recommendations.
Drilldowns were not needed yet, so it's fine to disable them now.
If they become needed, we can reenable them and reevaluate the memory cost, which might be fixed by then.

dagguh committed:
QUICK-224: Rerun the CircleCI IT, cache by Git commit

Use Git commit hash as the cache key to reuse the results during the rerun.

mminns committed:
QUICK-224: Run CircleCI IT on schedule or manually

mminns committed:
QUICK-224: Fix the XL dataset download URI

dagguh commented:
* Pointed to Bamboo CI
* Undid the VU reduction
* Rebased and squashed
